### DISCUSSION TITLE
fcitx-dbus issue

### DISCUSSION THREAD

**Discussion by GZYangKui** [fcitx/fcitx5#806] [Q&A]
I am currently docking with the FCITX dbus related interface, but I cannot find the relevant development documents on the official website. There is also a poor lack of information on Google. May I ask where I can obtain the corresponding development documents

**Comment by wengxt** [comment]
What kind of documentation you want?

The dbus interface provided by fcitx are considered as private api so there's no public document. You'll need to read code.

InputContext1 used by im module
https://github.com/fcitx/fcitx5/tree/master/src/frontend/dbusfrontend

Controller1 used mostly by fcitx5-configtool
https://github.com/fcitx/fcitx5/tree/master/src/module/dbus

If you are questioning the dbus wrapper in fcitx-utils/dbus, check codedocs.xyz/fcitx/fcitx5/ or read the doxygen style comment.
